function/registry/registry.go

```diff
@@ -155,42 +159,55 @@ func NewFilter(name string, summary func([]float64) float64, ascending bool) fun
 			if err != nil {
 				return nil, err
 			}
+			recentDuration := context.Timerange.Duration()
+			if len(arguments) == 3 {
+				recentDuration, err = function.EvaluateToDuration(arguments[2], context)
+				if err != nil {
+					return nil, err
+				}
+			}
 			// Round to the nearest integer.
 			count := int(countFloat + 0.5)
 			if count < 0 {
 				return nil, fmt.Errorf("expected positive count but got %d", count)
 			}
-			result := filter.FilterBy(list, count, summary, ascending)
+			if recentDuration < 0 {
```

Comment:
I think it's endpoint-inclusive, so this should mean "filter using only the very last point in the series."

I don't think it's very obvious but it should work.